app/gui/src/model/suggestion_database.rs

```diff
@@ -596,4 +668,203 @@ mod test {
+            module:             "TestProject.TestModule".to_string(),
+            arguments:          vec![],
+            return_type:        "TestAtom".to_string(),
+            documentation:      None,
+            documentation_html: None,
+            external_id:        None,
+        };
+        let entry2 = SuggestionEntry::Method {
+            name:               "create_process".to_string(),
+            module:             "Standard.Builtins.Main".to_string(),
+            self_type:          "Standard.Builtins.Main.System".to_string(),
+            arguments:          vec![],
+            return_type:        "Standard.Builtins.Main.System_Process_Result".to_string(),
+            documentation:      None,
+            documentation_html: None,
+            external_id:        None,
+        };
+        fn db_entry(id: SuggestionId, suggestion: SuggestionEntry) -> SuggestionsDatabaseEntry {
+            SuggestionsDatabaseEntry { id, suggestion }
+        }
+        let id1 = 1;
+        let id2 = 2;
+        let response = language_server::response::GetSuggestionDatabase {
+            entries:         vec![db_entry(id1, entry1), db_entry(id2, entry2)],
+            current_version: 1,
+        };
+        let db = SuggestionDatabase::from_ls_response(response);
+
+        // Modify the database contents by applying an update event.
+        let entry1_modification = Box::new(SuggestionsDatabaseModification {
+            arguments:          vec![],
+            module:             Some(FieldUpdate::set("NewProject.NewModule".to_string())),
+            self_type:          None,
+            return_type:        None,
+            documentation:      None,
+            documentation_html: None,
+            scope:              None,
+        });
+        let entry3 = SuggestionEntry::Module {
+            module:             "local.Unnamed_6.Main".to_string(),
+            documentation:      None,
+            documentation_html: None,
+            reexport:           None,
+        };
+        let update = SuggestionDatabaseUpdatesEvent {
+            updates:         vec![
+                entry::Update::Modify {
+                    id:           id1,
+                    external_id:  None,
+                    modification: entry1_modification,
+                },
+                entry::Update::Remove { id: id2 },
+                entry::Update::Add { id: 3, suggestion: entry3 },
+            ],
+            current_version: 2,
+        };
+        db.apply_update_event(update);
+
+        // Check the results of `lookup_by_fully_qualified_name` after the update.
+        expect_no_lookup(&db, "TestProject.TestModule.TextAtom");
+        expect_lookup(&db, "NewProject.NewModule.TextAtom");
+        expect_no_lookup(&db, "Standard.Builtins.Main.System.create_process");
+        expect_lookup(&db, "local.Unnamed_6.Main");
+    }
+
+    #[test]
+    fn lookup_by_fully_qualified_name_after_db_update_reuses_id() {
```

Comment:
As I've written in previous PR review - this thing is big and does not have docs explaining what it is testing. Reading 50-100 lines to understadn what test does is a bad idea.